src/ImageProcessor.Web/Caching/DiskCache.cs

```diff
@@ -211,64 +211,62 @@ public override Task TrimCacheAsync()
 
             this.ScheduleCacheTrimmer(token =>
             {
-                string rootDirectory = Path.GetDirectoryName(this.CachedPath);
-
+                var rootDirectory = Path.GetDirectoryName(this.CachedPath);
                 if (rootDirectory != null)
                 {
-                    // Jump up to the parent branch to clean through the cache.
-                    // UNC folders can throw exceptions if the file doesn't exist.
-                    IEnumerable<string> directories = SafeEnumerateDirectories(validatedAbsoluteCachePath).Reverse();
-
+                    // Jump up to the parent branch to clean through the cache
+                    // UNC folders can throw exceptions if the file doesn't exist
+                    var directories = SafeEnumerateDirectories(validatedAbsoluteCachePath).Reverse().ToList();
                     foreach (string directory in directories)
                     {
-                        if (!Directory.Exists(directory))
-                        {
-                            continue;
-                        }
-
                         if (token.IsCancellationRequested)
                         {
                             break;
                         }
 
-                        IEnumerable<FileInfo> files = Directory.EnumerateFiles(directory)
-                                                               .Select(f => new FileInfo(f))
-                                                               .OrderBy(f => f.CreationTimeUtc);
-                        int count = files.Count();
-
-                        foreach (FileInfo fileInfo in files)
+                        try
                         {
-                            if (token.IsCancellationRequested)
-                            {
-                                break;
-                            }
-
-                            try
+                            var files = Directory.EnumerateFiles(directory).Select(f => new FileInfo(f)).OrderBy(f => f.CreationTimeUtc).ToList();
+                            var count = files.Count;
```

Comment:
The previous `Count()` and foreach would enumerate the files twice, so converting it to a list first will ensure it's executed on the same line and only does it once.